### PR TITLE
Trajectory interface

### PR DESCRIPTION
**Pull request by fmauch**
Adds the trajectory interface from the [beta-testing](https://github.com/UniversalRobots/Universal_Robots_ROS_Driver/tree/beta-testing) branch.

There are a couple of things still missing (e.g. proper documentation, config- and launchfile integration)

### CODE REVIEW

File: ur_robot_driver/src/hardware_interface.cpp [hunk 1]
```diff
@@ -1185,23 +1178,24 @@ void HardwareInterface::cancelInterpolation()
...
   {
     case urcl::control::TrajectoryResult::TRAJECTORY_RESULT_SUCCESS:
     {
+      final_state = hardware_interface::ExecutionState::SUCCESS;
       ROS_INFO_STREAM("Forwarded trajectory finished successful.");
       break;
     }
     case urcl::control::TrajectoryResult::TRAJECTORY_RESULT_CANCELED:
     {
+      final_state = hardware_interface::ExecutionState::SUCCESS;
```
Comment (stefanscherzinger) [inline ur_robot_driver/src/hardware_interface.cpp:1192]:
```suggestion
      final_state = hardware_interface::ExecutionState::PREEMPTED;
```

File: ur_robot_driver/src/hardware_interface.cpp [hunk 2]
```diff
@@ -1185,23 +1178,24 @@ void HardwareInterface::cancelInterpolation()
...
       ROS_INFO_STREAM("Forwarded trajectory execution preempted by user.");
       break;
     }
     case urcl::control::TrajectoryResult::TRAJECTORY_RESULT_FAILURE:
     {
+      final_state = hardware_interface::ExecutionState::SUCCESS;
```
Comment (stefanscherzinger) [inline ur_robot_driver/src/hardware_interface.cpp:1198]:
```suggestion
      final_state = hardware_interface::ExecutionState::ABORTED;
```

File: ur_robot_driver/src/hardware_interface.cpp (hunk 1 above)
Comment (fmauch) [reply ur_robot_driver/src/hardware_interface.cpp]: Ah, this is where this came from :-D fixed in b59a729

File: ur_robot_driver/src/hardware_interface.cpp (hunk 2 above)
Comment (fmauch) [reply ur_robot_driver/src/hardware_interface.cpp]: Fixed in b59a729

**Comment by fmauch** [comment]
I'd like to add tests before merging -> converting to draft

**Pull request merged**